### PR TITLE
Add Python BigQuery pipeline for processing gcloud logs

### PR DESCRIPTION
**Pull request by Amar3tto**
Pipeline added: Google Cloud log processing with Apache Beam and Dataflow
IO: BigQuery IO
SDK Language: Python
Screenshot from 2023-12-08 18-33-02
Run instruction:
- Create BigQuery tables: input, output and DLQ.
- Go to [Log Router](https://console.cloud.google.com/logs/router) and route logs to the input BigQuery table (e.g. only logs for PubSub).
- run `python gcloud_logs_filter_with_dlq.py --runner="DataflowRunner" --project="your-project" --region="your-region" --input_table="your-input-table" --output_table="your-output-table" --dlq_table="your-dlq-table"`.
- Perform some actions (e.g. create/delete PubSub topic) so the logs will be routed to the input table.
- Check output BigQuery table.

### CODE REVIEW

**Review by olehborysevych — APPROVED** [review]
LGTM

**Comment by liferoad** [comment]
Can you fix the format and lint errors? @Amar3tto

File: Python/bigquery/gcloud_logs_filter_with_dlq.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+            split_result[PROCESSED_TAG]
+            | "Filter by Severity" >> Filter(filter_by_severity)
+            | "Map to Row for output" >> Map(to_row_for_output)
+            | "Write to BigQuery" >> WriteToBigQuery(
+                options.output_table,
+                schema=output_table_schema
+            )
+        )
+
+        # Write unprocessed messages to DLQ
```
Comment (Abacn) [inline Python/bigquery/gcloud_logs_filter_with_dlq.py:108]: This is a custom "dlq" example, not BigQueryIO's DLQ though. Is this what we need? CC: @johnjcasey

Comment (johnjcasey) [reply Python/bigquery/gcloud_logs_filter_with_dlq.py:108]: This is accurate for today. There is not yet a built in DLQ for BQ, and we don't have it implemented for cross language yet.

Comment (johnjcasey) [reply Python/bigquery/gcloud_logs_filter_with_dlq.py:108]: Also,  this doesn't look like it is handling errors writing to BQ as such, this is more the notion of "splitting" records into two separate streams

**Comment by johnjcasey** [comment]
This LGTM from a code perspective

**Pull request merged**